cmake/modules/GNUInstallPackageDir.cmake

```diff
@@ -0,0 +1,33 @@
+# Mimick `GNUInstallDirs` for one more install directory, the one where
+# project's installed cmake subdirs go.
+
+# These functions are internal functions vendored in from GNUInstallDirs (with
+# new names), so we don't depend on unstable implementation details. They are
+# also simplified to only handle the cases we need.
+#
+# The purpose would appear to be making `CACHE PATH` vars in a way that
+# bypasses the legacy oddity that `-D<PATH>` gets canonicalized, despite
+# non-canonical `CACHE PATH`s being perfectly valid.
+
+macro(_GNUInstallPackageDir_cache_convert_to_path var description)
```

Comment:
Initially the GNU name made my concerned about licensing, but this is vendored from upstream LLVM right? I guess that is fine. I don't know the policy for cmake, but it seems these files should have the usual LLVM license header, which would have made this less concerning.

Reply:
Do we need to replicate llvm/cmake/Modules/GNUInstallPackageDir.cmake rather than just import it?